File: backend/BankStatementConverter.Domain/Statements.cs
namespace BankStatementConverter.Domain;

public class BankStatement : Entity
{
    public Guid BankAccountId { get; set; }
    public BankAccount BankAccount { get; set; } = null!;
    public Guid CreatedBy { get; set; }
    public string OriginalFileName { get; set; } = "";
    public string StorageKey { get; set; } = "";
    public string FileHash { get; set; } = "";
    public long FileSize { get; set; }
    public int PageCount { get; set; }
    public string PdfType { get; set; } = "";
    public string Status { get; set; } = "UPLOADED";
    public string Message { get; set; } = "";
    public string RawText { get; set; } = "";
    public string Currency { get; set; } = "";
    public DateOnly? PeriodStart { get; set; }
    public DateOnly? PeriodEnd { get; set; }
    public decimal? OpeningBalance { get; set; }
    public decimal? ClosingBalance { get; set; }
    public Guid? ProfileId { get; set; }
    public Guid? ValidatedBy { get; set; }
    public DateTime? ValidatedAt { get; set; }
    public DateTime? ArchivedAt { get; set; }
    public DateTime? ProcessingStartedAt { get; set; }
    public Guid Version { get; set; } = Guid.NewGuid();
    public ICollection<BankTransaction> Transactions { get; set; } = [];
    public ICollection<StatementExport> Exports { get; set; } = [];
    public ICollection<ProcessingHistory> History { get; set; } = [];
}
public class BankTransaction : Entity
{
    public Guid BankStatementId { get; set; }
    public int Position { get; set; }
    public DateOnly? TransactionDate { get; set; }
    public DateOnly? ValueDate { get; set; }
    public string Reference { get; set; } = "";
    public string Description { get; set; } = "";
    public decimal? Debit { get; set; }
    public decimal? Credit { get; set; }
    public decimal? Balance { get; set; }
    public string Currency { get; set; } = "";
    public string RawText { get; set; } = "";
    public decimal? ConfidenceScore { get; set; }
    public bool IsValidated { get; set; }
}
public class BankStatementProfile : Entity
{
    public Guid BankId { get; set; }
    public Bank Bank { get; set; } = null!;
    public string Name { get; set; } = "";
    public string Code { get; set; } = "";
    public int Version { get; set; } = 1;
    public bool IsActive { get; set; }
    public string DetectionText { get; set; } = "";
    public string TransactionPattern { get; set; } = "";
    public string DateFormat { get; set; } = "dd/MM/yyyy";
    public string DecimalSeparator { get; set; } = ",";
    public string ThousandsSeparator { get; set; } = " ";
    public bool OcrRequired { get; set; }
}
public class ExportTemplate : Entity
{
    public string Name { get; set; } = "";
    public string Code { get; set; } = "";
    public string Type { get; set; } = "CUSTOM_CSV";
    public string Encoding { get; set; } = "UTF-8";
    public string Delimiter { get; set; } = ";";
    public string DateFormat { get; set; } = "dd/MM/yyyy";
    public string DecimalSeparator { get; set; } = ",";
    public int Decimals { get; set; } = 2;
    public bool IncludeHeader { get; set; } = true;
    public bool IsSystem { get; set; }
    public bool IsActive { get; set; } = true;
    public Guid Version { get; set; } = Guid.NewGuid();
    public ICollection<ExportTemplateField> Fields { get; set; } = [];
}
public class ExportTemplateField : Entity
{
    public Guid ExportTemplateId { get; set; }
    public string SourceField { get; set; } = "";
    public string OutputField { get; set; } = "";
    public int Position { get; set; }
    public bool Required { get; set; }
    public string DefaultValue { get; set; } = "";
}
public class StatementExport : Entity
{
    public Guid BankStatementId { get; set; }
    public Guid ExportTemplateId { get; set; }
    public ExportTemplate ExportTemplate { get; set; } = null!;
    public Guid CreatedBy { get; set; }
    public string FileName { get; set; } = "";
    public string StorageKey { get; set; } = "";
    public string FileHash { get; set; } = "";
    public long FileSize { get; set; }
    public string TemplateSnapshot { get; set; } = "";
    public string Status { get; set; } = "EXPORTED";
}
public class ProcessingHistory : Entity
{
    public Guid BankStatementId { get; set; }
    public Guid UserId { get; set; }
    public string Action { get; set; } = "";
    public string Status { get; set; } = "";
    public string Message { get; set; } = "";
}

File: frontend/Import-pdf-excel-master/src/services/statements.ts
import { API_URL, getSession } from './api';
export type Statement = { id:string; bankAccountId:string; clientId:string; client:string; bank:string; bankId:string; account:string; originalFileName:string; currency:string; status:string; transactionCount:number; createdAt:string; archivedAt:string|null; version:string; periodStart:string|null; periodEnd:string|null };
export type Transaction = { transactionDate:string|null; valueDate:string|null; reference:string; description:string; debit:number|null; credit:number|null; balance:number|null };
export type Field = {sourceField:string; outputField:string; position:number; required:boolean; defaultValue:string};
export type Template = {id:string; name:string; code:string; type:string; encoding:string; delimiter:string; dateFormat:string; decimalSeparator:string; decimals:number; includeHeader:boolean; isActive:boolean; version:string; fields:Field[]};
export type Detail = Statement & {message:string; pageCount:number; pdfType:string; openingBalance:number|null; closingBalance:number|null; validatedAt:string|null; profileId:string|null; rawText:string; overlappingPeriod:boolean; transactions:Transaction[]; check:{totalDebit:number;totalCredit:number;calculatedBalance:number|null;difference:number|null;issues:{code:string;message:string;row:number|null}[]}; history:{id:string;action:string;status:string;message:string;createdAt:string}[]; exports:{id:string;fileName:string;fileSize:number;createdAt:string}[]};
export const statusLabel:Record<string,string>={UPLOADED:'PDF importé',QUEUED:'En attente',ANALYZING:'Analyse',OCR_PROCESSING:'Lecture OCR',UNKNOWN_FORMAT:'Format inconnu',REVIEW_REQUIRED:'À vérifier',VALIDATED:'Validé',EXPORTED:'Exporté',ARCHIVED:'Archivé',EXTRACTION_FAILED:'Échec extraction',OCR_FAILED:'Échec OCR'};
export async function download(path:string,name:string) {
 const response=await fetch(API_URL+path,{headers:{Authorization:`Bearer ${getSession()?.token??''}`}});
 if(!response.ok){if(response.status===401)window.dispatchEvent(new Event('session-expired'));throw new Error('Téléchargement impossible ou accès refusé.');}
 const url=URL.createObjectURL(await response.blob());const a=document.createElement('a');a.href=url;a.download=name;document.body.append(a);a.click();a.remove();setTimeout(()=>URL.revokeObjectURL(url),10000);
}
